packages/core/helper-plugin/src/components/Table/index.js

```diff
@@ -0,0 +1,438 @@
+  Tbody,
+  Td,
+  Tooltip,
+  IconButton,
+  Thead,
+  Tr,
+  BaseCheckbox,
+  VisuallyHidden,
+  Loader,
+  Table as DSTable,
+} from '@strapi/design-system';
+import { useIntl } from 'react-intl';
+import { Trash } from '@strapi/icons';
+
+import useQueryParams from '../../hooks/useQueryParams';
+import { useTracking } from '../../features/Tracking';
+import ConfirmDialog from '../ConfirmDialog';
+import SortIcon from '../../icons/SortIcon';
+import EmptyStateLayout from '../EmptyStateLayout';
+
+/* -------------------------------------------------------------------------------------------------
+ * Context
+ * -----------------------------------------------------------------------------------------------*/
+
+const TableContext = React.createContext(null);
+
+const useTableContext = () => {
+  const context = React.useContext(TableContext);
+
+  if (!context) {
+    throw new Error('useTableContext must be used within a TableProvider');
+  }
+
+  return context;
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * ActionBar
+ * -----------------------------------------------------------------------------------------------*/
+
+const ActionBar = ({ children }) => {
+  const { formatMessage } = useIntl();
+  const { selectedEntries } = useTableContext();
+
+  if (!selectedEntries.length > 0) return null;
+
+  return (
+    <Flex paddingBottom={4} gap={3}>
+      <Typography variant="epsilon" textColor="neutral600">
+        {formatMessage(
+          {
+            id: 'content-manager.components.TableDelete.label',
+            defaultMessage: '{number, plural, one {# entry} other {# entries}} selected',
+          },
+          { number: selectedEntries.length }
+        )}
+      </Typography>
+      {children}
+    </Flex>
+  );
+};
+
+ActionBar.propTypes = {
+  children: PropTypes.node.isRequired,
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * BulkDeleteButton
+ * -----------------------------------------------------------------------------------------------*/
+
+const BulkDeleteButton = ({ onConfirmDeleteAll }) => {
+  const { selectedEntries, setSelectedEntries } = useTableContext();
+  const { formatMessage } = useIntl();
+  const [showConfirmDeleteAll, setShowConfirmDeleteAll] = useState(false);
+  const [isConfirmButtonLoading, setIsConfirmButtonLoading] = useState(false);
+  const { trackUsage } = useTracking();
+
+  const handleConfirmDeleteAll = async () => {
+    try {
+      setIsConfirmButtonLoading(true);
+      await onConfirmDeleteAll(selectedEntries);
+      setIsConfirmButtonLoading(false);
+      handleToggleConfirmDeleteAll();
+      setSelectedEntries([]);
+    } catch (err) {
+      setIsConfirmButtonLoading(false);
+      handleToggleConfirmDeleteAll();
+    }
+  };
+
+  const handleToggleConfirmDeleteAll = () => {
+    if (!showConfirmDeleteAll) {
+      trackUsage('willBulkDeleteEntries');
```

Comment:
Shouldn't this be a content manager specific tracking event?

Reply:
That's a good question. I'll have to double check how it was before, wouldn't we want to track bulk delete every it's used 🤔 ... Actually is bulk delete a CM specific thing?  I assumed there were other places it was used. I'll check, but if it is just the CM that would be great to remove all the bulk delete stuff from the helper plugin.

Reply:
Yeah you're right this is specific to the CM so I'll remove it here and we are tracking it in the BulkActionButtons